packages/@angular/cli/tasks/serve.ts

```diff
@@ -31,7 +31,15 @@ export default Task.extend({
     if (projectConfig.project && projectConfig.project.ejected) {
       throw new SilentError('An ejected project cannot use the build command anymore.');
     }
-    rimraf.sync(path.resolve(this.project.root, outputPath));
+
+    const contentBase = path.resolve(this.project.root, outputPath);
+    if (serveTaskOptions.useDist) {
+      if(!fs.existsSync(contentBase)) {
+        throw new SilentError('Dist directory not found');
```

Comment:
> I'd rather use ng build first to build EXACTLY what will be deployed then run the unit tests and e2e tests against what will be deployed. This is a pretty standard thing to do.


Good point.

At the moment the only way to do this is to run `e2e` with the same `build` options as the ones passed to get files for distribution (since `serve` inherits `build` options, and `e2e` inherits `serve` options).